### DISCUSSION TITLE
Adding a camera off a robot

### DISCUSSION THREAD

**Discussion by karl-exwzd** [cyberbotics/webots#3793] [Q&A]
I'm struggling with setting up an environment where I want to have a robot arm and a camera. So far, I've only seen examples of cameras attached to robots, but I would like to have a static camera instead.

For example: the UR robots have a `toolSlot` where end-effectors, cameras and other sensors can be added, but the sensors themselves cannot be added to the general scene or to a transform child, so as to remain static...

I was successful in defining a new abstract robot and add a child camera to that... but that feels wrong.

Is there a way to do this? Is this meant to be like that?

**Comment by omichel** [comment]
Yes, should should attach the camera to the root Robot node instead of putting it into a slot (which usually is at the end of the arm). This will require that you modify the proto file of the UR robot to add a `baseSlot` in addition to the `toolSlot`. This `baseSlot` should be located in the Robot node itself at the first level of children to lie at the basis of the robot and hence it won't move with the arm joints. Then you can orientate and translate this camera as much as you need to set it wherever you want relatively to the robot basis.
